Refactored item statuses into separate methods with validations

ziadoz opened the pull request:
https://trello.com/c/Q6HnRpgs/

I've split the status methods out, as it doesn't really make sense that they're set together. I've also changed the methods so that they only take valid statuses (it's Fusions' job to work out which product status maps to which Pangaea status). And I've added in some validation to ensure that the values given are correct, plus more tests, docblocks etc.

gabriel403 commented on tests/ItemsTest.php:
```diff
@@ -71,6 +74,22 @@ public function testItemsXml()
+    }
+
+    /**
+     * @expectedException \Pangaea\PangaeaException
```
Test the message too

gabriel403 commented:
Apart from comment :+1:

weshooper commented on src/Item.php:
```diff
@@ -3,13 +3,71 @@
-    const SPEC_DEFAULT_END_DATE  = '2049-12-31'; // as specified by Walmart
-    const VALID_ATTRIBUTE_GROUPS = ['Product', 'Compliance', 'MarketInProduct', 'MarketInOffer', 'Offer'];
+    /**
+     * Specification Default End Date (as specified by Walmart).
```
Don't need the leading 'Specification'

weshooper commented on src/Item.php:
```diff
@@ -64,17 +122,39 @@ public function setDates($start, $end = null)
+     */
+    public function setLifecycleStatus($status)
+    {
+        $status = mb_strtoupper($status);
```
Not sure if really matters a great deal since I can't imagine anyone else using the library anytime soon, but I'd been sticking with *not* using the `mb_` prefixed functions, contrary to inside Fusions, as (I think at least?) it requires an extension. Perhaps either mention that in the README, or use `strtoupper` for this simple case?

weshooper commented:
Other than minor comments, looks good! :+1:

gabriel403 commented on src/Item.php:
```diff
@@ -3,13 +3,71 @@
+     *
+     * @var mixed
+     */
     private $sku, $upc, $title, $shortDescription, $longDescription, $taxCode, $assets, $attributes, $brand, $itemLogistics;
```
`There MUST NOT be more than one property declared per statement.`

gabriel403 replied:
Not relevant to this pr I know, just thought I'd bring it up incase you decided to do it as well :D

ziadoz replied:
Is that a PSR warning? I didn't see anything on Semaphore about it.

I'd prefer each one declared separately and documented. I'll do it too if people want me to?

gabriel403 replied:
it's in the spec

weshooper replied:
Oops, that was myH^H^H^H^H^H^H *may* have been my fault! :grimacing:

ziadoz commented:
I've made some final amendments to this: 

- Documented requirements in `Readme` (PHP version, mb string extension).
- Added docblocks for each internal class property and moved them onto a line each.
- Made stray protected method private.
- Updated unit tests to test exception messages.

weshooper commented:
:+1:
